lib/util/host-rules.ts

```diff
@@ -8,13 +8,15 @@ import { parseUrl, validateUrl } from './url';
 
 let hostRules: HostRule[] = [];
 
-interface LegacyHostRule {
+export interface LegacyHostRule {
   hostName?: string;
   domainName?: string;
   baseUrl?: string;
+  host?: string;
+  endpoint?: string;
 }
 
-function migrateRule(rule: LegacyHostRule & HostRule): HostRule {
+export function migrateRule(rule: LegacyHostRule & HostRule): HostRule {
```

Comment:
this should be move to migration in a major release

Reply:
I don't totally get it.
Do you mean that we should move this to the custom migration file and handle the migration there & remove it from the `add` fn, which will lead to app breaking for some users.

Or just move the function `migrateRule` to tha custom migration file?

Reply:
move completely, but we need to wait until the on prem renovate was refactored to not use the add method directly.

the hosted app is already changed to a new backend, which no longer deep imports any renovate file.

Reply:
Can we already log a warning if the `add` function needs to migrate?

Reply:
What kind of long do you mean? For the app user I think, warning them the `add` fn will be modified soon and can't be used with `LegacyHostRules` fileds ie. multiple hosts.

Reply:
logger.warn(`Unmigrated host rule found`) or something like that, so that we can at least be aware before we remove it